### ISSUE TITLE
[prefer-static-string-properties] ability to excluded [title] from the list

### ISSUE THREAD

**Issue by maartentibau** [angular-eslint/angular-eslint#2380]
**Description and reproduction of the issue**

I think this ESLint rule is very useful, but it might be helpful to add the possibility to add some extra exclusions. For example in many Angular projects and `input` is being created with the name `title`. 

Now this creates sometimes a bit of an odd situation, because when you supply the title like this

```
<my-component title="This is my title" />
```

When you `hover` on this component you will get a `title` context menu that is standard behaviour in browsers.  There are ways to hide this with `CSS` I think, but that feels a bit overkill.

The solution to this is just passing the value like this

```
<my-component [title]="'This is my title'" />
```

But of course now the ESLint rule throws an error, so it would be helpful if it would be possible to pass some extra exclusions to this rule in the form of the actual names of properties which can be ignored.

![Image](https://github.com/user-attachments/assets/17c8ce43-f211-4689-be95-f6cd0804ba5e)

https://stackblitz.com/edit/angular-eslint-repros-fyfxqg2t?file=src%2Fmain.ts

**Versions**

| package                                  | version |
| ---------------------------------------- | ------- |
| `@angular-eslint/eslint-plugin-template` | `19.3.0` |
| `@angular-eslint/template-parser`        | `19.3.0` |
| `@typescript-eslint/parser`              | `19.3.0` |
| `ESLint`                                 | `9.16.0` |
| `node`                                   | `20.19.Z` |

```sh
# Please run `npx ng version` in your project and paste the full output here:


```

**Comment by Harpush** [comment]
I would add that a lint rule to enforce such cases to be with bindings can be useful too.

**Comment by reduckted** [comment]
> I would add that a lint rule to enforce such cases to be with bindings can be useful too.

Whatever the solution ends up being for the `prefer-static-string-properties` rule, I think this is the really important part. If you add an "ignore list" to `prefer-static-string-properties`, nothing would stop you from using a static string, and then you end up with the situation that was trying to be avoided in the first place.

**Comment by maartentibau** [comment]
> > I would add that a lint rule to enforce such cases to be with bindings can be useful too.
> 
> Whatever the solution ends up being for the `prefer-static-string-properties` rule, I think this is the really important part. If you add an "ignore list" to `prefer-static-string-properties`, nothing would stop you from using a static string, and then you end up with the situation that was trying to be avoided in the first place.

Well, personally that’s the whole point. If someone still wants to use a static string OR it’s actually wanted, for example like on a anchor tag you do not want to prohibit this.

The only thing is that this is just standard browser behavior, which can be altered by using CSS (i think) or by using the square brackets.

**Comment by Harpush** [comment]
I think  allow + deny is a good idea. So it enforces using or not using

**Comment by maartentibau** [comment]
> * if you don't have control, an `eslint-disable` comment is a perfectly fine way to handle this...
> 
> I think in general folks are too averse to disable comments for odd exceptions to rules. They are _always_ going to happen in codebases of reasonable size and complexity, lint rules are never going to be able to capture everything.
>

Thanks @JamesHenry for giving your view on this.

I think you are right that folks are too averse to use the `disable comments`, the reason for this (personally) is that I think it pollutes the code base. So I (personally) only use this when I really need to. I think I'm not alone in this, would be interesting to know why developers do not like to use the `disable` comments 😄.

In this case I think that the name `title` is so commonly used that it's kinda hard to come up with something else. I'm also sure that in a lot of cases developers do not even know of this `side-effect` and only get notified about this after user testing or E2E.

That being said... if I would need to count the amount of `disable` statements I would need to write to work around this issue, it would be a huge amount. So I looking at this from a DX point of view, it would make my life as a developer easier.

If adding the ability is not an option, is it possible to create a custom extension or plugin to overwrite the default behaviour?

**Comment by JamesHenry** [comment]
@maartentibau The third valid option is that you can of course always write your own rule that most precisely fits your needs, and we provide utilities to help you do that. That's the other thing that developers don't do as much as I would like 😄

**Comment by JamesHenry** [comment]
If we are going to make this change now is the time to do it as the major version is going to change this week and breaking changes can go in

**Comment by JamesHenry** [comment]
Please can someone craft a full proposal for the end state they want to see here?

**Comment by maartentibau** [comment]
> [@maartentibau](https://github.com/maartentibau) The third valid option is that you can of course always write your own rule that most precisely fits your needs, and we provide utilities to help you do that. That's the other thing that developers don't do as much as I would like 😄

OK, totally fine with that.  
Time to dive into this, I guess 🙂 

https://egghead.io/set-up-a-custom-es-lint-rule~kq49h

**Comment by NateRadebaugh** [comment]
If I have an input called `title` I normally use this host binding to unset the title in the DOM:

```ts
@HostBinding("attr.title") __title = ""; // Remove browser title/tooltip
```

**Comment by maartentibau** [comment]
I'm not 100% sure, but I think this could work. Don't know how long this rule already exists.

https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin-template/docs/rules/prefer-static-string-properties.md

But since the `ignore` option got added (https://github.com/angular-eslint/angular-eslint/commit/7cf27ebb0aeb6a7c577d0bebb40b02d080cd8c2d) it seems that is just what I need 😄 

Thank you @mistrykaran91 🙏 

Final result:

```
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error",
      {
        "ignore": [
          "title"
        ]
      }
    ]
  }
}
```
